bindgen/private/bindgen.bzl

```diff
@@ -129,26 +129,46 @@ def _rust_bindgen_impl(ctx):
 
     # Configure Clang Arguments
     args.add("--")
-    args.add_all(include_directories, before_each = "-I")
-    args.add_all(quote_include_directories, before_each = "-iquote")
-    args.add_all(system_include_directories, before_each = "-isystem")
-    args.add_all(ctx.attr.clang_flags)
 
-    cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
+    cc_toolchain, feature_configuration = find_cc_toolchain(ctx = ctx)
+    compile_variables = cc_common.create_compile_variables(
+        cc_toolchain = cc_toolchain,
+        feature_configuration = feature_configuration,
+        include_directories = cc_lib[CcInfo].compilation_context.includes,
+        quote_include_directories = cc_lib[CcInfo].compilation_context.quote_includes,
+        system_include_directories = cc_lib[CcInfo].compilation_context.system_includes,
+        user_compile_flags = ctx.attr.clang_flags,
+    )
+    compile_flags = cc_common.get_memory_inefficient_command_line(
+        feature_configuration = feature_configuration,
+        action_name = CPP_COMPILE_ACTION_NAME,
+        variables = compile_variables,
+    )
+
+    # Parse known arguments from the list of potential compile flags
+    open_arg = False
+    for arg in compile_flags:
+        if open_arg:
+            args.add(arg)
+            open_arg = False
+            continue
+
+        if arg in ctx.attr.clang_flags:
```

Comment:
This filtering seems a little odd? If someone has specified a duplicate flag, that seems fine (and the order may matter) and if it's a problem, they should probably remove it from the `clang_flags` attr?

Reply:
Yeah, I’m not super worried about duplicates. And again, I was hoping to use the toolchain to dictate order. In an ideal world I would be able to only get flags from a user defined list of features but that doesn’t seem possible.